File: app/src/main/java/com/example/product_tracker/ui/gallery/GalleryFragment.java
package com.example.product_tracker.ui.gallery;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.product_tracker.adapter.ProductAdapter;
import com.example.product_tracker.example.ProductDataSource;
import com.example.product_tracker.model.Product;
import com.example.product_tracker.R;
import com.example.product_tracker.databinding.FragmentGalleryBinding;

import java.util.List;

public class GalleryFragment extends Fragment {
    private String productType;
    private FragmentGalleryBinding binding; // Declare the binding variable

    public GalleryFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            productType = getArguments().getString("productType");
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentGalleryBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        RecyclerView productRecyclerView = root.findViewById(R.id.recyclerView);
        productRecyclerView.setLayoutManager(new GridLayoutManager(getContext(), 3));

        ProductAdapter productAdapter = new ProductAdapter(getProductList(), getContext());
        productRecyclerView.setAdapter(productAdapter);

        return root;
    }

    private List<Product> getProductList() {
        List<Product> productList = ProductDataSource.getProductList(productType);
        return productList;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/example/product_tracker/ui/gallery/GalleryFragment.kt
package com.example.product_tracker.ui.gallery

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import com.example.product_tracker.adapter.ProductAdapter
import com.example.product_tracker.databinding.FragmentGalleryBinding
import com.example.product_tracker.example.ProductDataSource
import com.example.product_tracker.model.Product

class GalleryFragment : Fragment() {
    private var productType: String? = null
    private var binding: FragmentGalleryBinding? = null // Declare the binding variable
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            productType = requireArguments().getString("productType")
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentGalleryBinding.inflate(inflater, container, false)
        val root: View = binding!!.root
        val productRecyclerView = binding!!.recyclerView
        productRecyclerView.layoutManager = GridLayoutManager(context, 3)
        val productAdapter = context?.let { ProductAdapter(productList, it) }
        productRecyclerView.adapter = productAdapter
        return root
    }

    private val productList: ArrayList<Product>
        get() = ProductDataSource.getProductList(productType)

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}